bin/runtime-common/src/messages.rs

```diff
@@ -104,6 +104,17 @@ pub trait ChainWithMessageLanes {
 	type MessageLaneInstance: Instance;
 }
 
+/// This chain that has `message-lane` and `call-dispatch` modules.
+pub trait ThisChainWithMessageLanes: ChainWithMessageLanes {
+	/// Are we accepting any messages to the given lane?
+	fn is_outbound_lane_enabled(lane: &LaneId) -> bool;
+
+	/// Maximal number of pending (not yet delivered) messages at this chain.
+	///
+	/// Any messages over this limit, will be rejected.
+	fn maximal_pending_messages_at_outbound_lane() -> MessageNonce;
```

Comment:
> Because these checks may be lane-specific - i.e. for some kind of sudo calls we may have a separate lane without any limits.

Given what you said, shouldn't the method then get `lane: &LaneId` parameter?

Reply:
No, I has commented specifically that our implementation of `MessageLaneVerifier` assumes that all lanes are identical. There are other checks (i.e. some lanes may accept messages with zero fee) that are not lane-specific in our implementation. So making this one lane-specific makes no sense

Reply:
Ah, I didn't notice that the check is actually being done in some verifier, so we can skip the check (or alter it) in case we don't want that.

Although, now that I think about it, perhaps it's better to use `frame_support::traits::Filter<LaneId>` for enablement (given there is some tooling around it) and just harcode the max pending messages in the verifier implementation (as a const)?

Reply:
> Although, now that I think about it, perhaps it's better to use frame_support::traits::Filter<LaneId> for enablement (given there is some tooling around it) and just harcode the max pending messages in the verifier implementation (as a const)?

This could be changed, but I feel that it'll just bring more code lines without any notable benefits. You may still use whatever you feel is better in the implementation - `Filter`s, `Get`s, constants or whatever.